department-of-veterans-affairs/caseflow
# Pamatya/appeals 30887

Resolves [APPEALS-30887](https://jira.devops.va.gov/browse/APPEALS-30887)

# Description
1. Added a new component called -> NonCompReportFilter, which wraps Searchable Dropdrown with RHF's Controller. 
2. Included NonCompReport Filter into report page. 
3. Disabled 2 button's -> Clear Filter and Generate task Report on initial load and activate it only when Report type dropdown is selected.
4. When Clear Filter is click it resets the dropdown's value to default.
5. Added a story for report page.
6. Added a Jest and feature test.

## Acceptance Criteria
- [ ] Drop-down creation to select between required filter: Type of Report: ‘Event / Action’ or ‘Status’. 
- [ ] ‘Generate task report’ button will not be enabled until Report type is selected. 
- [ ] 'Clear Filters' Button will not be enabled until Report type is selected.
- [ ] 'Clear Filters' button should reset the report type filter

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Login as VHA Admin user -> if VHA admin user is not present then either create one using following code:
```
user = create(:user)
vha_org = VhaBusinessLine.singleton
OrganizationsUser.make_user_admin(user, vha_org )
```
3. Goto decision_reviews/vha/report
4. Notice you are able to see Report type dropdown and both button disabled.
5. make a selection in the dropdown and notice both button are enabled now. 
6. Click on the Clear filter button, it should reset the dropdown to default value i.e select and buttons should be disabled. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/2018718e-8fa6-4436-b6da-c1fb19a313a3)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/c2f40a6d-5f62-44db-ab25-b4c12092832a)



## Storybook Story
*For Frontend (Presentation) Components*
* [x ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)


# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
